Source/MethodCall.cs

```diff
@@ -85,7 +85,7 @@ internal class TypeEqualityComparer : IEqualityComparer<Type>
 	{
 		public bool Equals(Type x, Type y)
 		{
-			return y.IsAssignableFrom(x);
+			return y.IsAssignableFrom(x) || y.IsAnyType();
 		}
 
 		public int GetHashCode(Type obj)
```

Comment:
Please be aware that the proposed change to `Equals` breaks `GetHashCode`: If `a.Equals(b)`, then `a.GetHashCode() == b.GetHashCode()` should also be true. Therefore, if every type is considered equal to `It.AnyType`, then it follows that every type should have the same hash code as `It.AnyType`. That is, `GetHashCode` should return a constant. Which renders it pretty much useless for hashed containers.

Fortunately, at the moment, this argument is rather academic. This implementation of `IEqualityComparer<Type>` is currently only used together with `Enumerable.SequenceEqual`, which doesn't use `GetHashCode` at all. In order to stop future contributions from using this equality comparer for scenarios in which it would not perform well, please change `GetHashCode` to the following:

```csharp
int IEqualityComparer<Type>.GetHashCode(Type obj) => throw new NotSupportedException();
```

... and add a brief comment why it throws.